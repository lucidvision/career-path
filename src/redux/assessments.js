const SET_ASSESSMENT = 'SET_ASSESSMENT'

export function setAssessment (assessment) {
  return {
    type: SET_ASSESSMENT,
    assessment
  }
}

const initialState = {
  assessment: []
}

export default function assessments (state = initialState, action) {
  switch (action.type) {
    case SET_ASSESSMENT :
      return {
        ...state,
        assessment: action.assessment
      }
    default :
      return state
  }
}
