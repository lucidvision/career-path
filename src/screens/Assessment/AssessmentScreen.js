import React, { Component } from 'react'
import { Alert, AsyncStorage, StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import { IconButton } from 'components'
import { Entypo } from '@expo/vector-icons'
import _ from 'lodash'

class AssessmentScreen extends Component {
  static navigationOptions = {
    title: 'Career Path'
  }
  state = {
    result: []
  }
  async componentDidMount () {
    try {
      const result = await AsyncStorage.getItem('assessmentResult')
      const resultArray = JSON.parse(result)
      this.setState({
        result: resultArray
      })
    } catch (error) {
      Alert(error)
    }
  }
  handleResultPressed = () => {
    this.props.navigation.navigate('Result', {name: 'Results', jobs: this.state.result})
  }
  handleAssessmentPressed = () => {
    this.props.navigation.navigate('Questionnaire')
  }
  render () {
    return (
      <View style={styles.container}>
        {!_.isEmpty(this.state.result) &&
          <TouchableOpacity onPress={this.handleResultPressed} style={styles.card}>
            <Text>{`You have ${this.state.result.length} job matches!`}</Text>
          </TouchableOpacity>}
        <IconButton onPress={this.handleAssessmentPressed} title='Start'>
          <Entypo name='text-document' size={25} />
        </IconButton>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center'
  },
  card: {
    padding: 10,
    borderColor: 'black',
    borderRadius: 10,
    borderWidth: 2,
    marginBottom: 10
  }
})

export default AssessmentScreen
