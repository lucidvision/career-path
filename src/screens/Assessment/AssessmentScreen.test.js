import React from 'react'
import { shallow } from 'enzyme'
import { AssessmentScreen } from './AssessmentScreen'
import { setAssessment } from 'redux/assessments'

const createAssessmentScreen = (override) => {
  const props = {
    assessment: {
      name: 'Doctor'
    },
    ...override
  }

  return shallow(<AssessmentScreen {...props} />)
}

describe('AssessmentScreen', () => {
  const component = createAssessmentScreen()

  it('renders correctly', () => {
    expect(component).toMatchSnapshot()
  })

  describe('componentDidMount', () => {
    jest.mock('AsyncStorage', () => {
      return {
        getItem: () => {
          return Promise.resolve('{"name": "Doctor"}')
        }
      }
    })

    it('dispatches assessment result', done => {
      const dispatchMock = jest.fn()
      const component = createAssessmentScreen({dispatch: dispatchMock})
      component.instance().componentDidMount()
      setTimeout(() => {
        expect(dispatchMock).toHaveBeenCalledWith(setAssessment({name: 'Doctor'}))
        done()
      })
    })
  })

  describe('handleResultPressed', () => {
    it('navigates to results screen', () => {
      const navMock = jest.fn()
      const component = createAssessmentScreen({navigation: {navigate: navMock}})
      component.find('TouchableOpacity').props().onPress()
      expect(navMock).toHaveBeenCalledWith('Result', {jobs: {name: 'Doctor'}, name: 'Results'})
    })
  })

  describe('handleAssessmentPressed', () => {
    it('navigates to assessment screen', () => {
      const navMock = jest.fn()
      const component = createAssessmentScreen({navigation: {navigate: navMock}})
      component.find('IconButton').props().onPress()
      expect(navMock).toHaveBeenCalledWith('Questionnaire')
    })
  })
})
