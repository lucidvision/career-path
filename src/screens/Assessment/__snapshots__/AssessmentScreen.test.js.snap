// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssessmentScreen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "borderColor": "black",
        "borderRadius": 10,
        "borderWidth": 2,
        "marginBottom": 10,
        "padding": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      You have undefined job matches!
    </Text>
  </TouchableOpacity>
  <IconButton
    onPress={[Function]}
    title="Start"
  >
    <Icon
      allowFontScaling={false}
      name="text-document"
      size={25}
    />
  </IconButton>
</View>
`;
